presto-tests/src/main/java/com/facebook/presto/tests/QueryAssertions.java

```diff
@@ -204,6 +204,102 @@ private static void assertQuery(
         }
     }
 
+    public static void assertQuery(
```

Comment:
this method looks like a giant copy-paste; I suggest to generalize the original source by introducing an ExpectedQueryRunner interface and adding expectedSession parameter:

```
    private interface ExpectedQueryRunner
    {
        MaterializedResult execute(Session session, @Language("SQL") String query, List<? extends Type> resultTypes);
    }

    private static ExpectedQueryRunner toExpectedQueryRunner(H2QueryRunner h2QueryRunner)
    {
        return (session, query, types) -> h2QueryRunner.execute(session, query, types);
    }

    private static ExpectedQueryRunner toExpectedQueryRunner(QueryRunner queryRunner)
    {
        return (session, query, types) -> queryRunner.execute(session, query);
    }

    private static void assertQuery(
            QueryRunner actualQueryRunner,
            Session session,
            @Language("SQL") String actual,
            H2QueryRunner h2QueryRunner,
            @Language("SQL") String expected,
            boolean ensureOrdering,
            boolean compareUpdate,
            Optional<Consumer<Plan>> planAssertion)
    {
        assertQuery(actualQueryRunner, session, actual, toExpectedQueryRunner(h2QueryRunner), session, expected, ensureOrdering, compareUpdate, planAssertion);
    }

    private static void assertQuery(
            QueryRunner actualQueryRunner,
            Session session,
            @Language("SQL") String actual,
            ExpectedQueryRunner expectedQueryRunner,
            Session expectedSession,
            @Language("SQL") String expected,
            boolean ensureOrdering,
            boolean compareUpdate,
            Optional<Consumer<Plan>> planAssertion)
    {
      ...
```

Then, introduce two new executeQuery methods:

```

    public static void assertQuery(
            QueryRunner queryRunner,
            Session session,
            @Language("SQL") String actual,
            Session expectedSession,
            String expected)
    {
        assertQuery(queryRunner, session, actual, toExpectedQueryRunner(queryRunner), expectedSession, expected, false, false, Optional.empty());
    }

    public static void assertQuery(
            QueryRunner queryRunner,
            Session session,
            @Language("SQL") String actual,
            Session expectedSession)
    {
        assertQuery(queryRunner, session, actual, toExpectedQueryRunner(queryRunner), expectedSession, actual, false, false, Optional.empty());
    }
```